service/graph/graph.go

```diff
@@ -25,20 +25,27 @@ import (
 	"go.opentelemetry.io/collector/service/pipelines"
 )
 
-// Settings holds configuration for building builtPipelines.
+// Settings holds configuration for building a Graph.
 type Settings struct {
+	// Telemetry specifies the telemetry settings.
 	Telemetry component.TelemetrySettings
+	// BuildInfo provides collector start information.
 	BuildInfo component.BuildInfo
 
-	ReceiverBuilder  *receiver.Builder
+	// ReceiverBuilder is a helper struct that given a set of Configs and Factories helps with creating receivers.
+	ReceiverBuilder *receiver.Builder
+	// ProcessorBuilder is a helper struct that given a set of Configs and Factories helps with creating processors.
 	ProcessorBuilder *processor.Builder
-	ExporterBuilder  *exporter.Builder
+	// ExporterBuilder is a helper struct that given a set of Configs and Factories helps with creating exporters.
+	ExporterBuilder *exporter.Builder
+	// ConnectorBuilder is a helper struct that given a set of Configs and Factories helps with creating connectors.
 	ConnectorBuilder *connector.Builder
 
 	// PipelineConfigs is a map of component.ID to PipelineConfig.
 	PipelineConfigs pipelines.Config
```

Comment:
I would separate this in a different argument to `Build`. See https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/extensions/extensions.go#L125

Reply:
Makes sense, done